### ISSUE TITLE
Laz demo and dual-monitor Windows box

### ISSUE THREAD

**Issue by the-Arioch** [castle-engine/castle-engine#373]
D:\DelphiProjects\Libs\Games\castle-engine\examples\lazarus\model_3d_with_2d_controls\mainf.pas

This window has "old skool" setting `TForm.Position = poDesktopCenter`. This means in multi-monitor coniguration the form is set at the center of "virtual desktop". In 2-monitors case - it would be exactly the borderline BETWEEN two monitors.

Newer Delphi introduced `poScreenCenter` value and deprecated `poDesktopCenter` (but left it for legacy apps, dunno why).  VCL and LCL docs suggest replacing `poDesktopCenter` -> `poScreenCenter` in all projects.

Dunno about `TCastleWindowBase` if it has similar property. As for now non-LCL demos i tried worked ok.

**Comment by michaliskambi** [comment]
That would be weird behavior of `poDesktopCenter` (whoever would want to center forms this way?)... but indeed, searching the Internet it seems true: for Delphi http://www.delphigroups.info/2/78/444696.html, and for LCL it says "poDesktopCenter,   // center form on desktop (total of all screens)" in https://github.com/alrieckert/lazarus/blob/master/lcl/forms.pp .

I'll test on my multi-monitor setup at home later next week (I'm curious whether it is indeed so weird, in LCL, for both WinAPI and GTK) and fix if necessary.

**Comment by the-Arioch** [comment]
i think it is the same in LCL, both because Delphi (VCL) compatibility
and because LCL documentation (just like VCL) one documents that
option is obsolete.

as of why Delphi designers did so, instead of changing behavior of
poDesktopCenter, i agree there seems no good reason for it, and it is
confusing.

well, it is not the worst thing VCL designers did :-)

the  TModalResult in DFM debacle was much worse IMHO, i'd even say it f-up

**Comment by the-Arioch** [comment]
ping

test and fix :-)

trivial and sadly typical for Delphi legacy projects

**Comment by michaliskambi** [comment]
Fixed both our projects.

Note that neither of the 2 affected CGE examples was (ever) a Delphi project, they were created and used only with LCL. Still, it seems LCL suffers from the same deficiency as you describe.

I tested the problem now only on Linux with GTK backend, and can confirm it's a problem. Running LCL application with `poDesktopCenter`, with a small main form -> places this form weirdly at the left border of the right monitor. I guess that LCL actually tried to position it in the middle, but GTK fixed it -> placing it at the left border of the right monitor. Still a weird place, and definitely not what we had in mind.

`poScreenCenter` behaves as expected.
